addons/loyalty/models/loyalty_reward.py

```diff
@@ -233,6 +233,8 @@ def write(self, vals):
         if 'active' in vals:
             if vals['active']:
                 self.reward_product_id.action_unarchive()
+            elif not vals['active'] and self[0].reward_type == 'product':
+                self.discount_line_product_id.action_archive()
             else:
                 self.reward_product_id.action_archive()
```

Comment:
Shouldn't we always archive the `discount_line_product_id` instead of `reward_product_id`?

cc @dalaOdoo 

```suggestion
            if vals['active']:
                self.discount_line_product_id.action_unarchive()
            else:
                self.discount_line_product_id.action_archive()
```

Reply:
I am not too knowledgeable in this functionally so I don't know if there might be a scenario that we will have to archive the `reward_product_id` but as per as the scenarios I tested I agree with you. It makes more sense to archive `discount_line_product_id`